packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -833,3 +833,83 @@ module JsArray = {
     |> Js.Array.map(O.toExn("Warning: This should not have happened"))
   let filter = Js.Array.filter
 }
+
+module Duration = {
+  //Stores in Unix milliseconds
+  type t = float
+  let minute = Belt.Float.fromInt(60 * 1000)
+  let hour = Belt.Float.fromInt(60 * 60 * 1000)
+  let day = Belt.Float.fromInt(24 * 60 * 60 * 1000)
+  let year = Belt.Float.fromInt(24 * 60 * 60 * 1000) *. 365.25
+  let fromFloat = (f: float): t => f
+  let toFloat = (d: t): float => d
+  let fromMinutes = (h: float): t => h *. minute
+  let fromHours = (h: float): t => h *. hour
+  let fromDays = (d: float): t => d *. day
+  let fromYears = (y: float): t => y *. year
+  let toMinutes = (t: t): float => t /. minute
+  let toHours = (t: t): float => t /. hour
+  let toDays = (t: t): float => t /. day
+  let toYears = (t: t): float => t /. year
+
+  let toString = (t: t): string => {
+    let shouldPluralize = f => f != 1.0
+    let display = (f: float, s: string) =>
+      `${Float.with3DigitsPrecision(f)} ${s}${shouldPluralize(f) ? "s" : ""}`
+    let abs = Js.Math.abs_float(t)
+    if abs >= year {
+      display(t /. year, "year")
+    } else if abs >= day {
+      display(t /. day, "day")
+    } else if abs >= hour {
+      display(t /. hour, "hour")
+    } else if abs >= minute {
+      display(t /. minute, "minute")
+    } else {
+      Float.toFixed(t) ++ "ms"
+    }
+  }
+  let add = (t1: t, t2: t): t => t1 +. t2
+  let subtract = (t1: t, t2: t): t => t1 -. t2
+  let multiply = (t1: t, t2: float): t => t1 *. t2
+  let divide = (t1: t, t2: float): t => t1 /. t2
+}
+
+module Date = {
+  //The Rescript/JS implementation of Date is pretty mediocre. It would be good to improve upon later.
```

Comment:
Yea; I think this could come later. The first challenge is to come up with a high-level interface for users to easily work with time. Later we can improve the underlying implementation.